README.md

```diff
@@ -1,3 +1,253 @@
+	}
+
+	// use the environment source to get values from environment variables
+	envSrc := &env.Source{}
+
+	// the Config struct will be populated in the order in which the sources are
+	// passed in the Config function with increasing precedence. So the fileSrc value
+	// will overwrite the flagSet value if they both were to set for the same field
+	view, err := dials.Config(context.Background(), config, envSrc, flagSet, fileSrc)
+	if err != nil {
+		// error handling
+	}
+
+	// Config holds the populated config struct
+	Config := view.View().(*Config)
+	fmt.Printf("Config: %+v\n", Config)
+}
+```
+
+If the above code is run with the following YAML file (make sure to change the path in the code):
+
+``` yaml
+b: valueb
+val-3: false
+```
+
+and the following commands 
+
+``` 
+export VAL_2=5
+go run main.go --val-3
+```
+ 
+the output will be `Config: &{Val1:valueb Val2:5 Val3:true}`. Note that even when val-3 is defined in the yaml file and the file source takes precedence, 
+only the value from command line flag populates the config due to the special `dialsflag` tag. If you prefer the value for `Val3` be overwritten by other sources, then
+use the `dials` tag instead of the `dialsflag` tag
+
+
+
+### Watching file source
+If you wish to watch the config file and make updates to your configuration, use the watching source. This functionality is available in the `ez` package by passing the `true` boolean to the functions. The `WatchingSource` can be used when you want to further customize the configuration as well.
+
+``` go
+	// NewWatchSource also has watch options that can be passed to have the
+	// ability to use a ticker for polling, set a logger, and more
+	watchingFileSource, fsErr := file.NewWatchingSource(
+		"path/to/config", &yaml.Decoder{})
+
+	if fsErr != nil {
+		// error handling
+		return
+	}
+
+	// additional sources can be passed along with the watching file source and the
+	// precedence order will still be dictated by the order in which the sources are
+	// defined in the Config function.
+	view, err := dials.Config(context.Background(), config, watchingFileSource)
+	if err != nil {
+		// error handling
+	}
+
+	Config := view.View().(*Config)
+```
+
+### Source
+Source interface is implemented by different configuration sources that populate the configuration struct. Dials currently supports environment variables, command line flags, and config file sources. When `dials.Config` function is going through the different sources to extract the values, it calls the `Value` method on each of these sources. This allows for the logic of the source to be encapsulated while giving the application access to the values populated by each source. The `dials.Config` function composes the final config struct by overlaying the values from all the different sources and accounting for the precedence order. If you wish to define your own source, implement the `Source` interface and pass the source to the `dials.Config` function.
```

Comment:
We should probably break this section up into a few paragraphs. (and probably have a subsection about writing a `Source`). -- including a mention of when it's appropriate to delegate to a `Decoder` implementation.

We should probably also include a mention of pointerification in a "nitty-gritty" section or something, so it's clear how we actually implement overlaying.